src/ApacheOrcDotNet.OptimizedReader/Infrastructure/HttpByteRangeProvider.cs

```diff
@@ -35,31 +34,31 @@ public async Task<int> GetRangeAsync(Memory<byte> buffer, long position)
             var response = await _httpClient.SendAsync(request);
 
             if (!response.Content.Headers.ContentRange.Length.HasValue)
-                throw new InvalidOperationException("Range respose must include a length.");
+                throw new InvalidOperationException("Range response must include a length.");
 
-            return await DoReadAsync(await response.Content.ReadAsStreamAsync(), buffer);
+            return await (await response.Content.ReadAsStreamAsync()).ReadAsync(buffer);
```

Comment:
Let's split this into two lines.

Also see the previous note regarding the return value.